lte/gateway/c/core/oai/test/ngap/test_ngap_flows.cpp

```diff
@@ -1413,4 +1415,92 @@ TEST_F(NgapFlowTest, test_ue_notifications_from_amf) {
   bdestroy(ngap_initial_ue_msg);
 }
 
+TEST_F(NgapFlowTest, NgapHandleSctpDisconnection) {
+  MessageDef* sctp_message_p = NULL;
+  m5g_ue_description_t* ue_ref = NULL;
+  Ngap_InitialUEMessage_t* container;
+
+  unsigned char initial_ue_message_hexbuf[] = {
+      0x00, 0x0f, 0x40, 0x48, 0x00, 0x00, 0x05, 0x00, 0x55, 0x00, 0x02,
+      0x00, 0x01, 0x00, 0x26, 0x00, 0x1a, 0x19, 0x7e, 0x00, 0x41, 0x79,
+      0x00, 0x0d, 0x01, 0x22, 0x62, 0x54, 0x00, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x01, 0x2e, 0x04, 0xf0, 0xf0, 0xf0, 0xf0, 0x00,
+      0x79, 0x00, 0x13, 0x48, 0x22, 0x42, 0x65, 0x00, 0x00, 0x00, 0x01,
+      0x00, 0x22, 0x42, 0x65, 0x00, 0x00, 0x01, 0xe4, 0xf7, 0x04, 0x44,
+      0x00, 0x5a, 0x40, 0x01, 0x18, 0x00, 0x70, 0x40, 0x01, 0x00};
+
+  std::vector<MessagesIds> expected_Ids{NGAP_INITIAL_UE_MESSAGE,
+                                        NGAP_GNB_DEREGISTERED_IND};
+
+  // Verify sctp association is successful
+  EXPECT_EQ(ngap_handle_new_association(state, &peerInfo), RETURNok);
+  // Verify number of connected gNB's is 1
+  EXPECT_EQ(state->gnbs.num_elements, 1);
+
+  Ngap_NGAP_PDU_t decoded_pdu = {};
+  uint16_t length = sizeof(initial_ue_message_hexbuf) / sizeof(unsigned char);
+  bstring ngap_initial_ue_msg = blk2bstr(initial_ue_message_hexbuf, length);
+
+  // Check if the pdu can be decoded
+  ASSERT_EQ(ngap_amf_decode_pdu(&decoded_pdu, ngap_initial_ue_msg), RETURNok);
+
+  // check if initial UE message is handled successfully
+  EXPECT_EQ(ngap_amf_handle_message(state, peerInfo.assoc_id,
+                                    peerInfo.instreams, &decoded_pdu),
+            RETURNok);
+
```

Comment:
Can we put these lines above, which indicates that association is setup. Followed by sending Initial Ue Message which depicts the call flow and easier to follow.

Reply:
Addressed Review Comment.